packages/adaptivecpp/package.py

```diff
@@ -218,36 +218,82 @@ def filter_config_file(self):
         config_file_path = config_file_paths[0]
         with open(config_file_path) as f:
             config = json.load(f)
+        
+        # There may be a separate cuda config file
+        cuda_config = None
+        cuda_config_file_path = None
+        config_file_paths = filesystem.find(
+            self.prefix, ("acpp-cuda.json",)
+        )
+        if len(config_file_paths) > 0:
+            cuda_config_file_path = config_file_paths[0]
+            with open(cuda_config_file_path) as f:
+                cuda_config = json.load(f)
+
         # 1. Fix compiler: use the real one in place of the Spack wrapper
         config["default-cpu-cxx"] = self.compiler.cxx
         # 2. Fix stdlib: we need to make sure cuda-enabled binaries find
         #    the libc++.so and libc++abi.so dyn linked to the sycl
         #    ptx backend
-
+        
+        # Find the rpaths for cpp
+        rpaths_cpp = set()
         if "llvm" in self.spec:
-            rpaths = set()
             so_paths = filesystem.find(self.spec["llvm"].prefix, "libc++.so")
             if len(so_paths) != 1:
                 raise InstallError(
                     "concretized llvm dependency must provide a "
                     "unique directory containing libc++.so, "
                     "found: {0}".format(so_paths)
                 )
-            rpaths.add(path.dirname(so_paths[0]))
+            rpaths_cpp.add(path.dirname(so_paths[0]))
             so_paths = filesystem.find(self.spec["llvm"].prefix, "libc++abi.so")
             if len(so_paths) != 1:
                 raise InstallError(
                     "concretized llvm dependency must provide a "
                     "unique directory containing libc++abi.so, "
                     "found: {0}".format(so_paths)
                 )
-            rpaths.add(path.dirname(so_paths[0]))
+            rpaths_cpp.add(path.dirname(so_paths[0]))
+
+            # the omp llvm backend may link against the libomp.so in llvm
+            so_paths = filesystem.find(self.spec["llvm"].prefix, "libomp.so")
+            rpaths_cpp.add(path.dirname(so_paths[0]))
+        
+        if "llvm" in self.spec:
```

Comment:
Why is this a separate if-block from the above? They have the same condition.

Reply:
Thanks, fused these conditionals.